src/GitVersionTask/VersionAndBranchFinder.cs

```diff
@@ -27,17 +27,20 @@ public static VersionVariables GetVersion(string directory, Authentication authe
         var gitDir = GitDirFinder.TreeWalkForDotGitDir(directory);
         using (var repo = RepositoryLoader.GetRepo(gitDir))
         {
-            var ticks = DirectoryDateFinder.GetLastDirectoryWrite(directory);
-            var key = string.Format("{0}:{1}:{2}", repo.Head.CanonicalName, repo.Head.Tip.Sha, ticks);
+            var ticks = DirectoryDateFinder.GetLastDirectoryWrite(gitDir);
+            var key = string.Format("{0}:{1}:{2}",gitDir, repo.Head.CanonicalName, repo.Head.Tip.Sha);
```

Comment:
we dont want ticks in the key. they are used in the second step. a static dictionary is bascially a memory leak. keying on ticks is not great since we gradually add more to the dictionary on every change.